From 77ffa0dfe640642bc397b8cffc266347a9de3c80 Mon Sep 17 00:00:00 2001
From: Srujana Challa <schalla@marvell.com>
Date: Fri, 27 Nov 2020 18:31:58 +0530
Subject: [PATCH 0946/1921] octeontx2-af: Fix a loop issue in CPT FLR handler

This patch fixes a infinite loop issue with
'commit 4738cccaea30 ("octeontx2-af: Handle CPT function level reset")'

Signed-off-by: Srujana Challa <schalla@marvell.com>
Change-Id: I17bce1fb66f4e947285aa2aa712eb7e07ce05c09
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/41192
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
index ca7700aad0d3..c3c855c26ff1 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
@@ -557,7 +557,7 @@ static void cpt_lf_disable_iqueue(struct rvu *rvu, int blkaddr, int slot)
 		 * knowing that no more instructions will be scheduled
 		 * (INFLIGHT == 0) && (GWB < 40) && (GRB == 0 OR 40)
 		 */
-		if (((inprog && 0x1FF) == 0) &&
+		if (((inprog & 0x1FF) == 0) &&
 		    (((inprog >> 40) & 0xFF) < 40) &&
 		    ((((inprog >> 32) & 0xFF) == 0) ||
 		    (((inprog >> 32) & 0xFF) == 40)))
-- 
2.31.1

